tripy/tripy/frontend/tensor.py

```diff
@@ -205,51 +191,47 @@ def eval(self) -> Array:
         assert len(data) == 1, "Expects only one output from mlir_tensorrt.compiler executor"
         data = data[0]
         # Data is present now. Assign the underlying device type.
-        self.device = data.device
+        self.device = flat_ir.outputs[0].device
 
         Storage.build_internal([], [self.trace_tensor], data)
-        self.trace_tensor.shape = data.shape
         return data
 
-    def data(self) -> Array:
-        import tripy.common.datatype
-        from tripy.frontend.trace.ops.cast import cast
-
-        arr = self.eval()
-        if self.dtype in [tripy.common.datatype.float8, tripy.common.datatype.int4]:
-            arr = cast(Tensor(arr), tripy.common.datatype.float32).eval()
-        return arr
+    def tolist(self):
+        data_memref = self.eval()
+        if self.dtype in (datatype.float8, datatype.int4):
```

Comment:
Doesn't this need to be all types not natively supported by Python?

Reply:
Yes, added `bfloat16`.

Reply:
Updated, thanks